src/core/model/src/main/java/it/geosolutions/geostore/core/model/Resource.java

```diff
@@ -119,6 +120,10 @@ public class Resource implements Serializable, CycleRecoverable {
     @OneToMany(mappedBy = "resource", cascade = CascadeType.REMOVE, fetch = FetchType.LAZY)
     private List<SecurityRule> security;
 
+    @Transient private boolean canEdit;
+    @Transient private boolean canDelete;
+    @Transient private boolean canCopy;
```

Comment:
You’re mixing persistence fields with UI/permission flags in the same entity.
Consider documenting clearly that these fields are only set at runtime, or wrap the Resource in a “ResourceView” object so that the entity remains focused on business/persistence data, and UI permission logic lives elsewhere.

Reply:
You're right.
I've created a DTO for the resource+permission data, aligning it to the `Ext` naming already in place.
Thanks!